feat: add contribution guide for the docs (#6)
kubesaw/kubesaw.github.io: README.md

# KubeSaw Docs

This repository contains the documentation website code and Markdown source files for kubesaw.github.io.

## Contributing

The website is built using [mkdocs](https://www.mkdocs.org/getting-started/#getting-started-with-mkdocs) 

### Prerequisites

- python3
```shell
python3 --version
Python 3.11.5
```

> ### 1.  Create an isolated virtual environment

Reviewer: Should we mention that the contributor should first fork the repo (and clone it)?

Author: Good point, I usually give that for grant but better to be explicit ! Thanks!

Author: pls see : https://github.com/kubesaw/kubesaw.github.io/pull/6/commits/44c7c9c7d9987b308624aa4597afab10650fbf58

From the root folder of the repo run: 
```shell
python3 -m venv venv
``` 
NOTE: the second `venv` is the name of the folder containing the [virtual environment](https://docs.python.org/3/library/venv.html)

> Now you need to activate the virtual env so that all the dependencies will be installed only in here, and not on your global python intallation.

Reviewer: ```suggestion
Now you need to activate the virtual env so that all the dependencies will be installed only here, not on your global python installation.
```

Author: TY!

```shell
source venv/bin/activate
```

### 2. Install dependencies

```shell
pip install -r requirements.txt 
```

NOTE:
- add newly needed dependencies in the `requirements.txt` file

### 3. Run the local server

```shell

> mkdocs serve

Reviewer: When I run this command, I am getting:
```
Error: Config file 'mkdocs.yml' does not exist.
```

Do we need this config file or am I missing something?

# KubeSaw Docs

This repository contains the documentation website code and Markdown source files for kubesaw.github.io.

## Contributing

The website is built using [mkdocs](https://www.mkdocs.org/getting-started/#getting-started-with-mkdocs)

**NOTE**: To contribute changes, make sure you do that from your own fork!

### Prerequisites

- python3
```shell
python3 --version
Python 3.11.5
```

### 1.  Create an isolated virtual environment
From the root folder of the repo run: 
```shell
python3 -m venv venv
``` 

> NOTE: the second `venv` is the name of the folder containing the [virtual environment](https://docs.python.org/3/library/venv.html)

Reviewer 2: ```suggestion
**NOTE**: the second `venv` is the name of the folder containing the [virtual environment](https://docs.python.org/3/library/venv.html)
```

Now you need to activate the virtual env so that all the dependencies will be installed only here, not on your global python installation.

```shell
source venv/bin/activate
```

### 2. Install dependencies

```shell

> pip install -r requirements.txt 

Reviewer 2: I'm not sure what I did wrong:
```
/[~/kubesaw/kubesaw.github.io] (main)$ python3 --version
Python 3.11.8
/[~/kubesaw/kubesaw.github.io] (main)$ python3 -m venv venv
/[~/kubesaw/kubesaw.github.io] (main)$ source venv/bin/activate
/[~/kubesaw/kubesaw.github.io] (main)$ pip install -r requirements.txt 
ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'
```

Reviewer 2: never mind, I'm idiot - I tested it in the `main` branch, not in your PR :man_facepalming: 
I need anther coffee :coffee: or maybe two or three :sweat_smile:

```

> NOTE:
> - add newly needed dependencies in the `requirements.txt` file

Reviewer 2: ```suggestion
**NOTE**: add newly needed dependencies in the `requirements.txt` file
```

### 3. Run the local server

```shell

> mkdocs serve

Reviewer 2: this failed for me with:
```sh
Error: Config file 'mkdocs.yml' does not exist.
```
not sure what I'm missing now :thinking: 

```bash
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ python3 --version
Python 3.11.8
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ python3 -m venv venv
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ source venv/bin/activate
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ pip install -r requirements.txt 
Requirement already satisfied: mkdocs in ./venv/lib64/python3.11/site-packages (from -r requirements.txt (line 1)) (1.6.1)
Requirement already satisfied: mkdocs-material in ./venv/lib64/python3.11/site-packages (from -r requirements.txt (line 2)) (9.5.47)
Requirement already satisfied: click>=7.0 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (8.1.7)
Requirement already satisfied: ghp-import>=1.0 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (2.1.0)
Requirement already satisfied: jinja2>=2.11.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (3.1.4)
Requirement already satisfied: markdown>=3.3.6 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (3.7)
Requirement already satisfied: markupsafe>=2.0.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (3.0.2)
Requirement already satisfied: mergedeep>=1.3.4 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (1.3.4)
Requirement already satisfied: mkdocs-get-deps>=0.2.0 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (0.2.0)
Requirement already satisfied: packaging>=20.5 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (24.2)
Requirement already satisfied: pathspec>=0.11.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (0.12.1)
Requirement already satisfied: pyyaml-env-tag>=0.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (0.1)
Requirement already satisfied: pyyaml>=5.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (6.0.2)
Requirement already satisfied: watchdog>=2.0 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (6.0.0)
Requirement already satisfied: babel~=2.10 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (2.16.0)
Requirement already satisfied: colorama~=0.4 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (0.4.6)
Requirement already satisfied: mkdocs-material-extensions~=1.3 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (1.3.1)
Requirement already satisfied: paginate~=0.5 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (0.5.7)
Requirement already satisfied: pygments~=2.16 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (2.18.0)
Requirement already satisfied: pymdown-extensions~=10.2 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (10.12)
Requirement already satisfied: regex>=2022.4 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (2024.11.6)
Requirement already satisfied: requests~=2.26 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (2.32.3)
Requirement already satisfied: python-dateutil>=2.8.1 in ./venv/lib64/python3.11/site-packages (from ghp-import>=1.0->mkdocs->-r requirements.txt (line 1)) (2.9.0.post0)
Requirement already satisfied: platformdirs>=2.2.0 in ./venv/lib64/python3.11/site-packages (from mkdocs-get-deps>=0.2.0->mkdocs->-r requirements.txt (line 1)) (4.3.6)
Requirement already satisfied: charset-normalizer<4,>=2 in ./venv/lib64/python3.11/site-packages (from requests~=2.26->mkdocs-material->-r requirements.txt (line 2)) (3.4.0)
Requirement already satisfied: idna<4,>=2.5 in ./venv/lib64/python3.11/site-packages (from requests~=2.26->mkdocs-material->-r requirements.txt (line 2)) (3.10)
Requirement already satisfied: urllib3<3,>=1.21.1 in ./venv/lib64/python3.11/site-packages (from requests~=2.26->mkdocs-material->-r requirements.txt (line 2)) (2.2.3)
Requirement already satisfied: certifi>=2017.4.17 in ./venv/lib64/python3.11/site-packages (from requests~=2.26->mkdocs-material->-r requirements.txt (line 2)) (2024.8.30)
Requirement already satisfied: six>=1.5 in ./venv/lib64/python3.11/site-packages (from python-dateutil>=2.8.1->ghp-import>=1.0->mkdocs->-r requirements.txt (line 1)) (1.17.0)
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ mkdocs serve
Error: Config file 'mkdocs.yml' does not exist.
```

Author: I think you did everything correctly. The problem is that the `mkdocs.yml` file was added in this PR https://github.com/kubesaw/kubesaw.github.io/pull/5 and not the current one 🤦‍♂️

Author: But now that I've merged that, I'm noticing that GH doesn't suggest me to update this PR 🤔 . I'm wondering why.

Author: Anyways , I've rebased the PR and tried again, it should be working now. Sorry for the confusion. Please let me know if it works for you. You do need to pull the new commits ( which will include the `mkdocs.yml` file that was missing initially ) 

As per the PR update button probably I'm missing some configuration in project.

Reviewer 2: > But now that I've merged that, I'm noticing that GH doesn't suggest me to update this PR 🤔 . I'm wondering why.

it's in the general settings of the project - I updated it.
Btw, it was also missing the branch protection for the `main` branch - I added it so it requires at least one review approval.

Author: thanks a lot for fixing it 🙏
